boards/raspberry_pi_pico_2/src/main.rs

```diff
@@ -0,0 +1,436 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright OxidOS Automotive 2025.
+
+//! Tock kernel for the Raspberry Pi Pico 2.
+//!
+//! It is based on RP2350SoC SoC (Cortex M33).
+
+#![no_std]
+// Disable this attribute when documenting, as a workaround for
+// https://github.com/rust-lang/rust/issues/62184.
+#![cfg_attr(not(doc), no_main)]
+#![deny(missing_docs)]
+
+use core::ptr::{addr_of, addr_of_mut};
+
+use capsules_core::virtualizers::virtual_alarm::VirtualMuxAlarm;
+use components::gpio::GpioComponent;
+use components::led::LedsComponent;
+use enum_primitive::cast::FromPrimitive;
+use kernel::component::Component;
+use kernel::hil::led::LedHigh;
+use kernel::platform::{KernelResources, SyscallDriverLookup};
+use kernel::scheduler::round_robin::RoundRobinSched;
+use kernel::syscall::SyscallDriver;
+use kernel::{capabilities, create_capability, static_init, Kernel};
+
+use rp2350::chip::{Rp2350, Rp2350DefaultPeripherals};
+use rp2350::clocks::{
+    AdcAuxiliaryClockSource, HstxAuxiliaryClockSource, PeripheralAuxiliaryClockSource, PllClock,
+    ReferenceAuxiliaryClockSource, ReferenceClockSource, SystemAuxiliaryClockSource,
+    SystemClockSource, UsbAuxiliaryClockSource,
+};
+use rp2350::gpio::{GpioFunction, RPGpio, RPGpioPin};
+use rp2350::resets::Peripheral;
+use rp2350::timer::RPTimer;
+#[allow(unused)]
+use rp2350::{xosc, BASE_VECTORS};
+
+mod io;
+
+mod flash_bootloader;
+
+/// Allocate memory for the stack
+#[no_mangle]
+#[link_section = ".stack_buffer"]
+pub static mut STACK_MEMORY: [u8; 0x3000] = [0; 0x3000];
```

Comment:
If this goes in after #4473, we should remove the `pub` here.

Reply:
#4473 was merged, so I removed the `pub`